locksmith/src/controllers/loginController.ts

```diff
@@ -0,0 +1,69 @@
+import { Request, Response } from 'express'
+import { ErrorTypes, generateNonce, SiweMessage } from 'siwe'
+import { logger } from '../logger'
+
+export class LoginController {
+  async login(request: Request, response: Response) {
+    try {
+      if (!request.body.message) {
+        response.status(422).json({
+          message: 'Expected prepareMessage object as body.',
+        })
+      }
+      const message = new SiweMessage(request.body.message)
+      const fields = await message.validate(request.body.signature)
+      if (fields.nonce !== request.session.nonce) {
+        logger.info(`${request.session} failed to validate.`)
+        response.status(422).json({
+          message: 'Invalid nonce',
+        })
+      }
+      request.session.siwe = fields
+      await request.session.save()
```

Comment:
Is that an express API?

Reply:
no, that is attached by the sessionMiddleware from iron.